### DISCUSSION TITLE
Create node/tree from string

### DISCUSSION THREAD

**Discussion by markhuge** [nvim-treesitter/nvim-treesitter#2820] [General]
Hi, I'm trying to implement a function for snippets that would accept the input of a go struct (as a string), and generate a snippet that implements a literal of that struct. Since this is happening inside of lua snippet code, it's not actually available as a node in the tree of the go file at the time the snippet is being generated and expanded. All I have available is a string containing code for the struct.

Is there anything that allows for creating an arbitrary node/tree for a specific language from a string containing code for that language? Should I be approaching this differently?

**Comment by theHamsta** [comment]
See `:h get_string_parser`. Most methods for tree-sitter accept either a string or a buffer (like creating a languagetree https://github.com/neovim/neovim/blob/1facad23473eb2d045fe77199b3b0b9fd2586895/runtime/lua/vim/treesitter/languagetree.lua#L18

You could have a look on telescope that uses tree-sitter string parsing a lot for its previews

**Reply by markhuge** [reply]
TIL github discussions are a thing.

`get_string_parser` is exactly what I was looking for. Thank you!
